homeassistant/helpers/entityfilter.py

```diff
@@ -19,6 +26,8 @@ def convert_filter(config: Dict[str, List[str]]) -> Callable[[str], bool]:
         config[CONF_INCLUDE_ENTITIES],
         config[CONF_EXCLUDE_DOMAINS],
         config[CONF_EXCLUDE_ENTITIES],
+        config.get(CONF_INCLUDE_ENTITY_GLOBS, []),
+        config.get(CONF_EXCLUDE_ENTITY_GLOBS, []),
```

Comment:
It looks like these are always passing though the schema and will always get defaults set, we should not use `.get`

`config[CONF_INCLUDE_ENTITY_GLOBS]`

Reply:
I see your point but this one concerns me a bit.  First when I make this change there are 16 failures in the `homekit` tests. I'll go through them but I assume the situation is similar to above, where it is creating the schema directly as a dictionary and thus things that would normally be defaulted at runtime are skipped. One test in the list is one I added so I can adjust that but the others were all pre-existing.
```
Results (2.97s):
       6 passed
      16 failed
         - tests/components/homekit/test_homekit.py:89 test_setup_min[pyloop]
         - tests/components/homekit/test_homekit.py:125 test_setup_auto_start_disabled[pyloop]
         - tests/components/homekit/test_homekit.py:321 test_homekit_add_accessory[pyloop]
         - tests/components/homekit/test_homekit.py:357 test_homekit_remove_accessory[pyloop]
         - tests/components/homekit/test_homekit.py:381 test_homekit_entity_filter[pyloop]
         - tests/components/homekit/test_homekit.py:415 test_homekit_entity_glob_filter[pyloop]
         - tests/components/homekit/test_homekit.py:456 test_homekit_start[pyloop]
         - tests/components/homekit/test_homekit.py:541 test_homekit_start_with_a_broken_accessory[pyloop]
         - tests/components/homekit/test_homekit.py:591 test_homekit_stop[pyloop]
         - tests/components/homekit/test_homekit.py:628 test_homekit_reset_accessories[pyloop]
         - tests/components/homekit/test_homekit.py:676 test_homekit_too_many_accessories[pyloop]
         - tests/components/homekit/test_homekit.py:710 test_homekit_finds_linked_batteries[pyloop]
         - tests/components/homekit/test_homekit.py:795 test_setup_imported[pyloop]
         - tests/components/homekit/test_homekit.py:913 test_homekit_uses_system_zeroconf[pyloop]
         - tests/components/homekit/test_homekit.py:940 test_homekit_ignored_missing_devices[pyloop]
         - tests/components/homekit/test_homekit.py:1019 test_homekit_finds_linked_motion_sensors[pyloop]
```
However my bigger concern with this change isn't the tests that are failing, its the ones I don't know about. When going through this code I noticed not everyone is really using entityfilter as I believe it was originally intended. Where they import `FILTER_SCHEMA` and then add that to their config. Since the `convert_filter` and `generate_filter` functions are not marked as private I found direct usage of them. This is why for instance there was no shared `include`/`exclude` schema between `logbook` and `recorder` prior to my change.

Since helpers are core I'm worried there will be custom components doing the same thing. And therefore if I do not default the new glob option here at runtime, they will fail on update. I also figured that since this function was intended for schema validation time as opposed to filter evaluation time using `get` here had a low cost but prevented inadvertent upstream issues.

What do you think? Am I right to be concerned here?

Reply:
I made `convert_filter` public in #34560 fairly recently, so probably only homekit is using it so that shouldn't be too much of a concern if it is only the `homekit` tests.

Its better to keep the input validation and update the tests. We can label this a breaking change to make sure custom components are aware.

Reply:
Ok well now I feel bad. This hid a legit minor bug. I didn't realize that when homekit was setup from a config entry it didn't use the schema at all and instead had its own logic for setting defaults when necessary in `async_setup_entry`. Small tweak and now all of them pass.